open-metadata-implementation/access-services/cognos/cognos-server/src/main/java/org/odpi/openmetadata/accessservices/cognos/assets/DatabaseContextHandler.java

```diff
@@ -0,0 +1,500 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright Contributors to the ODPi Egeria project. */
+package org.odpi.openmetadata.accessservices.cognos.assets;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.collections4.CollectionUtils;
+
+import org.odpi.openmetadata.accessservices.cognos.contentmanager.OMEntityDao;
+import org.odpi.openmetadata.accessservices.cognos.ffdc.CognosErrorCode;
+import org.odpi.openmetadata.accessservices.cognos.ffdc.exceptions.CognosRuntimeException;
+import org.odpi.openmetadata.accessservices.cognos.model.ResponseContainerDatabase;
+import org.odpi.openmetadata.accessservices.cognos.model.ResponseContainerDatabaseSchema;
+import org.odpi.openmetadata.accessservices.cognos.model.ResponseContainerModule;
+import org.odpi.openmetadata.accessservices.cognos.model.ResponseContainerSchemaTables;
+import org.odpi.openmetadata.accessservices.cognos.model.module.Column;
+import org.odpi.openmetadata.accessservices.cognos.model.module.DataSource;
+import org.odpi.openmetadata.accessservices.cognos.model.module.ForeignColumn;
+import org.odpi.openmetadata.accessservices.cognos.model.module.ForeignKey;
+import org.odpi.openmetadata.accessservices.cognos.model.module.Module;
+import org.odpi.openmetadata.accessservices.cognos.model.module.PrimaryKey;
+import org.odpi.openmetadata.accessservices.cognos.model.module.Table;
+import org.odpi.openmetadata.accessservices.cognos.model.module.TableItem;
+import org.odpi.openmetadata.accessservices.cognos.utils.Constants;
+
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.properties.instances.Classification;
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.properties.instances.EntityDetail;
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.properties.instances.InstanceProperties;
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.properties.instances.Relationship;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DatabaseContextHandler {
+
+
+	private static final Logger log = LoggerFactory.getLogger(DatabaseContextHandler.class);
+	private OMEntityDao omEntityDao;
+
+	public DatabaseContextHandler(OMEntityDao omEntityDao) {
+		this.omEntityDao = omEntityDao;
+	}
+	
+	/**
+	 * Set context of the execution.
+	 * Every public method should call the method setContext() first to set the context.
+	 */
+	void setContext() {
+		omEntityDao.setContext(new Throwable().getStackTrace()[1].getMethodName());
```

Comment:
This is an expensive way to get the method name - typically we set up a static final string in the REST services layer and pass it down - it means that the method name in externalised messages and exceptions is a method name that is meaningful to the consumer.

For logic/unexpected errors the stack trace is logged to the audit log in an EXCEPTION message.
For checked exceptions, the error handling values should have enough information to debug the problem without the stack trace.

Reply:
fixed